### PR TITLE
(MODULES-5459) Update Documentation

### PR DESCRIPTION
**Pull request by RandomNoun7**
This change brings the documentation in the type files into sync with
what was present in the README.md file. In some cases the text was
better in the type file and so the two sources were combined.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -252,15 +252,15 @@ Allows creation of a new IIS Application Pool and configuration of application p
 
 #### `ensure`
 
-Must be either 'present' or 'absent'. Present will ensure the application pool is created.
+Specifies whether an application pool should be present or absent. If `state` is not specified, the application pool will be created and left in the default started state.
 
 #### `name`
 
-Name of the application pool.
+The unique name of the ApplicationPool.
```
Comment (michaeltlombardi) [inline README.md]: Possibly rephrase this to:

> The name of the ApplicationPool. **Must** be unique.

File: README.md [hunk 2]
```diff
@@ -252,15 +252,15 @@ Allows creation of a new IIS Application Pool and configuration of application p
...
 
 #### `state`
 
-The state of the application pool. Must be either 'started' or 'stopped'.
+The state of the ApplicationPool. By default, a newly created application pool will be started
```
Comment (michaeltlombardi) [inline README.md]: Suggest re-including that the value must be either started or stopped in addition to the new information.

File: README.md [hunk 3]
```diff
@@ -503,7 +507,7 @@ Must be either 'present' or 'absent'.
 
 ##### `name`
 
-The name of the feature.
+The unique name of the feature to manage.
```
Comment (michaeltlombardi) [inline README.md]: As above, suggest clarifying that it must be unique as opposed to uniqueness being a intrinsic quality of the property.

ie, the difference between 'the unique name' and 'the name which must be unique' is semantically important.

File: README.md [hunk 4]
```diff
@@ -536,7 +540,7 @@ If 'stopped' is specified, then the site will be created and stopped.
 
 ##### `name`
 
-The name of the site.
+The Name of the IIS site. Used for uniqueness. Will set the target to this value if target is unset.
```
Comment (michaeltlombardi) [inline README.md]: Rephrase to clarify that it must be unique to keep in line with other notes.

File: README.md [hunk 5]
```diff
@@ -552,7 +556,17 @@ The protocols enabled for the site. If 'https' is specified, 'http' is implied.
 
 ##### `bindings`
 
-Specifies one or more bindings (protocol, address, port, and ssl certificate) for the site.
+The protocol, address, port, and ssl certificate bindings for a web site.
+
+The bindinginformation value should be in the form of the IPv4/IPv6 address or wildcard *, then the port, then the optional hostname separated by colons:  `(ip|\*):[1-65535]:(hostname)?`
```
Comment (michaeltlombardi) [inline README.md]: Suggest placing the wildcard in backticks to fix formatting and clarify.

File: lib/puppet/type/iis_application.rb [hunk 6]
```diff
@@ -27,7 +37,8 @@ def self.title_patterns
   end
 
   newparam(:applicationname, :namevar => true) do
-    desc "The name of the Application. The virtual path of an application is /<applicationname>"
+    desc "The name of the application. The virtual path of the application is
+          '/<applicationname>'."
```
Comment (michaeltlombardi) [inline lib/puppet/type/iis_application.rb:41]: Is this change to accommodate line length?

File: README.md (hunk 1 above)
Comment (RandomNoun7) [reply README.md]: Extremely agree.

File: README.md (hunk 2 above)
Comment (RandomNoun7) [reply README.md]: Agree to the fullest extent of the law.

**Review by michaeltlombardi — CHANGES_REQUESTED** [review]
A few minor changes and tweaks.

File: lib/puppet/type/iis_application.rb (hunk 6 above)
Comment (RandomNoun7) [reply lib/puppet/type/iis_application.rb:41]: Yes. For a while I was wrapping at only approximately  80 chars, but it got annoying to be in the business of making a hundred judgement calls about when to wrap and when not to wrap, so I switch to a programmatic 80 char wrap.

File: README.md [hunk 7]
```diff
@@ -536,7 +540,7 @@ If 'stopped' is specified, then the site will be created and stopped.
 
 ##### `name`
 
-The name of the site.
+The Name of the IIS site. Must be uniquene. Will set the target to this value if target is unset.
```
Comment (michaeltlombardi) [inline README.md]: Spelling - 'uniquene' should be 'unique'

**Review by michaeltlombardi — APPROVED** [review]
